src/main/java/uk/ac/ebi/eva/submission/controller/admin/AdminController.java

```diff
@@ -44,4 +51,59 @@ public ResponseEntity<?> markSubmissionStatus(@PathVariable("submissionId") Stri
+    public ResponseEntity<?> getSubmissionsbyStatus(@PathVariable("status") SubmissionStatus status) {
+        try {
+            ArrayList<Submission> submissions = (ArrayList<Submission>) submissionService.getSubmissionsByStatus(status);
+            return new ResponseEntity<>(submissions, HttpStatus.OK);
+        } catch (SubmissionDoesNotExistException ex) {
+            return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @Operation(summary = "Given a submission id, this endpoint updates the processing status of the submission to the one provided",
+            security = {@SecurityRequirement(name = "basicAuth")
+            })
+    @Parameters({
+            @Parameter(name="submissionId", description = "Id of the submission whose status needs to be updated",
+                    required = true, in= ParameterIn.PATH),
+            @Parameter(name="step", description = "The processing step of the submission",
+                    required = true, in= ParameterIn.PATH),
+            @Parameter(name="status", description = "The status of the processing step for this submission",
+                    required = true, in= ParameterIn.PATH)
+    })
+    @PutMapping("submission-process/{submissionId}/{step}/{status}")
+    public ResponseEntity<?> markSubmissionProcessStepAndStatus(@PathVariable("submissionId") String submissionId,
+                                                                @PathVariable("step") SubmissionProcessingStep step,
+                                                                @PathVariable("status") SubmissionProcessingStatus status) {
+        SubmissionProcessing submissionProc = this.submissionService.markSubmissionProcessStepAndStatus(submissionId, step, status);
+        return new ResponseEntity<>(submissionProc, HttpStatus.OK);
+    }
+
+
+    @Operation(summary = "This endpoint retrieves all the submissions of a specific status present in the database")
+    @Parameters({
+            @Parameter(name="step", description = "The processing step of the submission.",
+                    required = true, in= ParameterIn.PATH),
+            @Parameter(name="status", description = "The status of the submission processing step.",
+                    required = true, in= ParameterIn.PATH)
+    })
+    @GetMapping("submission-processes/{step}/{status}")
+    public ResponseEntity<?> getSubmissionsProcessingByStepAndStatus(
+            @PathVariable("step") SubmissionProcessingStep step,
+            @PathVariable("status") SubmissionProcessingStatus status) {
+        try {
+            ArrayList<SubmissionProcessing> submissions = (ArrayList<SubmissionProcessing>) submissionService.getSubmissionsByProcessingStepAndStatus(step, status);
+            return new ResponseEntity<>(submissions, HttpStatus.OK);
+        } catch (SubmissionDoesNotExistException ex) {
+            return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
+        }
```

Comment:
```suggestion
            List<SubmissionProcessing> submissions = submissionService.getSubmissionsByProcessingStepAndStatus(step, status);
            return new ResponseEntity<>(submissions, HttpStatus.OK);
```
this will not throw any exception

Reply:
Done

Reply:
Not sure if you forgot to push the changes, but it is still there.